src/main/java/com/redhat/lightblue/camel/LightblueProducer.java

```diff
@@ -22,7 +24,20 @@ public LightblueProducer(LightblueScheduledPollEndpoint endpoint) {
     public void process(Exchange exchange) throws Exception {
         LightblueRequest req = (AbstractLightblueDataRequest) exchange.getIn().getBody();
 
-        exchange.getOut().setBody(endpoint.getLightblueClient().data(req));
+        try {
+            LightblueResponse response = endpoint.getLightblueClient().data(req);
+            if (response.hasError()) {
+                throw new RuntimeException("LightblueProducer:Error returned in response: " + response.getText());
```

Comment:
> instead of setting exception on the exchange in two places, I decided to just throw an exception and setting it in the catch clause ( we will have to do that anyway)

Exception handling is expensive. I would rather we just set the exception rather than spend the cost of a short lived exception. And I don't think a RuntimeException adds any value to the stacktrace.